pkg/provisioner/reconciler/claim-reconciler/reconiler.go

```diff
@@ -259,15 +266,18 @@ func (r *ObjectBucketClaimReconciler) handleDeleteClaim(key client.ObjectKey) er
 		log.Error(nil, "got nil objectBucket, assuming deletion complete")
 		return nil
 	}
+	// see if obc delete is for a newly provisioned bkt vs an existing bkt
+	isNewBkt := r.internalClient.isNewBucketByOB(ob)
 
-	class, err := storageClassForOB(ob, r.internalClient)
-	if err != nil || class == nil {
-		return fmt.Errorf("error getting storageclass from OB %q", ob.Name)
+	// Call the provisioner's `Revoke` method for old (brownfield) buckets regardless of reclaimPolicy.
+	// Also call `Revoke` for new buckets with a reclaimPolicy other than "Delete".
+	if ob.Spec.ReclaimPolicy == nil {
```

Comment:
this is a pretty far out edge case.  worth checking?

Reply:
remove check. Internal integrity err only and probably a panic is the right result of this ptr is nil.